packages/react-charting/src/components/PieChart/Pie/Pie.tsx

```diff
@@ -28,23 +31,36 @@ export class Pie extends React.Component<IPieProps, {}> {
         innerRadius={this.props.innerRadius}
         outerRadius={this.props.outerRadius}
         color={`${this.colors(i)}`}
+        theme={this.props.theme}
       />
     );
   };
 
   public render(): JSX.Element {
     // const getClassNames = classNamesFunction<IPieProps, IPieStyles>();
-    const { pie, colors, data, width, height, chartTitle } = this.props;
+    const { pie, data, width, height, chartTitle, theme } = this.props;
 
-    this.colors = scale.scaleOrdinal().range(colors!);
+    const defaultColors: Array<string> = [];
+    if (data && !this.props.colors) {
+      for (let i = 0; i < data.length; i++) {
+        defaultColors.push(getNextColor(i, 0, theme?.isInverted));
+      }
+    }
+    const { colors = defaultColors } = this.props;
+
+    this.colors = scale.scaleOrdinal().range(colors.map(color => getColorFromToken(color)));
 
     const piechart = pie(data);
     const translate = `translate(${width / 2}, ${height / 2})`;
 
     return (
-      <svg width={width} height={height} aria-label={chartTitle}>
-        <g transform={translate}>{piechart.map((d: IArcData, i: number) => this.arcGenerator(d, i))}</g>
-      </svg>
+      <FocusZone direction={FocusZoneDirection.domOrder}>
+        <svg width={width} height={height} aria-label={chartTitle}>
+          <g className="arc" transform={translate}>
```

Comment:
why are you putting className arc in pie and className pie in arc?

Reply:
mistakenly did that, fixed now. Thanks for noticing.

Reply:
Now the arc className has disappeared from snapshots.

Reply:
So as far as I understand

in snapshots it has the following structure:
<svg> // Pie.tsx
<g> // Pie.tsx
<g> // Arc.tsx
<path> // Arc.tsx
</g>  // Arc.tsx
</g> // Pie.tsx
</svg> // Pie.tsx

Now the className="arc" (hardcoded string) was the in <g> inside Arc.tsx, but I can't find any place where this class is defined or what purpose it was serving. I replaced it with className={classNames.arc} and used that to define some styles, but when classes are defined like this, the snapshot doesn't capture any className but the JSON defining the styles itself.

So should I put back the className="arc" back there to make it reflect in the snapshots, and define my styles in some other way (using styles={{}} prop).

Reply:
It is fine if the className="arc" does not appear in snapshots but appear in actual rendered DOM.
This is needed for backward compatibility where someone might have used this class to define some custom styles of their own that we don't want to break.

Reply:
got it, Thanks for explaining. Added className "arc"